airflow/providers/hashicorp/common/vault_client.py

```diff
@@ -0,0 +1,458 @@
+        if auth_type == "azure":
+            if not azure_resource:
+                raise VaultError("The 'azure' authentication type requires 'azure_resource'")
+            if not azure_tenant_id:
+                raise VaultError("The 'azure' authentication type requires 'azure_tenant_id'")
+        if auth_type == "radius":
+            if not radius_host:
+                raise VaultError("The 'radius' authentication type requires 'radius_host'")
+            if not radius_secret:
+                raise VaultError("The 'radius' authentication type requires 'radius_secret'")
+
+        self.kv_engine_version = kv_engine_version if kv_engine_version else 2
+        self.url = url
+        self.auth_type = auth_type
+        self.kwargs = kwargs
+        self.token = token
+        self.auth_mount_point = auth_mount_point
+        self.mount_point = mount_point if mount_point else 'secret'
+        self.username = username
+        self.password = password
+        self.key_id = key_id
+        self.secret_id = secret_id
+        self.role_id = role_id
+        self.kubernetes_role = kubernetes_role
+        self.kubernetes_jwt_path = kubernetes_jwt_path
+        self.gcp_key_path = gcp_key_path
+        self.gcp_keyfile_dict = gcp_keyfile_dict
+        self.gcp_scopes = gcp_scopes
+        self.azure_tenant_id = azure_tenant_id
+        self.azure_resource = azure_resource
+        self.radius_host = radius_host
+        self.radius_secret = radius_secret
+        self.radius_port = radius_port
+
+    @staticmethod
+    def get_default_mount_point(auth_type: str, kv_engine_version: int = 2) -> str:
+        # pylint: disable=too-many-return-statements
+        """
+        Returns default mount point.
+
+        :param auth_type: Authentication Type for Vault. Available Values are in the VALID_AUTH_TYPES.
+        :type auth_type: str
+        :param kv_engine_version: Select the version of the engine to run (``1`` or ``2``, default: ``2``).
+        :type kv_engine_version: int
+        :rtype: str
+        :return: Default mount point for the auth_type and KV engine version
+
+        """
+        if auth_type == "approle":
+            return "approle"
+        if auth_type == "aws_iam":
+            return aws.DEFAULT_MOUNT_POINT
+        if auth_type == "azure":
+            return azure.DEFAULT_MOUNT_POINT
+        if auth_type == "gcp":
+            return gcp.DEFAULT_MOUNT_POINT
+        if auth_type == "github":
+            return github.DEFAULT_MOUNT_POINT
+        if auth_type == "kubernetes":
+            return kubernetes.DEFAULT_MOUNT_POINT
+        if auth_type == "ldap":
+            return ldap.DEFAULT_MOUNT_POINT
+        if auth_type == "radius":
+            return radius.DEFAULT_MOUNT_POINT
+        if auth_type == "userpass":
+            return userpass.DEFAULT_MOUNT_POINT
+        return kv_v1.DEFAULT_MOUNT_POINT if kv_engine_version == 1 else kv_v2.DEFAULT_MOUNT_POINT
```

Comment:
Do we need this defaults? They are already the defaults even if don't specify

Reply:
The whole method can be removed now.